apps/ejabberd/src/mod_http_notification.erl

```diff
@@ -0,0 +1,81 @@
+%%%----------------------------------------------------------------------
+%%% File    : mod_notify.erl
+%%% Author  : Baibossynv Valery <baibossynov.valery@gmail.com>
+%%% Purpose : Message passing via http
+%%% Created : 16 Dec 2015 by Baibossynv Valery <baibossynov.valery@gmail.com>
+%%%----------------------------------------------------------------------
+
+-module(mod_http_notification).
+-author("baibossynov.valery@gmail.com").
+-behaviour(gen_mod).
+
+%% API
+-export([start/2, stop/1, on_user_send_packet/3,should_make_req/4]).
+
+-include("ejabberd.hrl").
+-include("jlib.hrl").
+
+-define(DEFAULT_HTTP_POOL_SIZE, 5).
+-define(DEFAULT_HTTP_HOST, "http://localhost").
+-define(DEFAULT_PREFIX_PATH, <<"/">>).
+
+start(Host, _Opts) ->
+  HttpHost = gen_mod:get_module_opt(Host, ?MODULE, host, ?DEFAULT_HTTP_HOST),
+  PoolSize = gen_mod:get_module_opt(Host, ?MODULE, pool_size, ?DEFAULT_HTTP_POOL_SIZE),
+  ChildMods = [fusco],
+  ChildMF = {fusco, start_link},
+  ChildArgs = {for_all, [HttpHost, []]},
+  {ok, _} = supervisor:start_child(ejabberd_sup,
+    {{http_notification_sup, Host},
+      {cuesport, start_link,
+        [pool_name(Host), PoolSize, ChildMods, ChildMF, ChildArgs]},
+      transient, 2000, supervisor, [cuesport | ChildMods]}),
+  ejabberd_hooks:add(user_send_packet, Host, ?MODULE, on_user_send_packet, 100),
+  ok.
+
+stop(Host) ->
+  Ch = {http_notification_sup, Host},
+  supervisor:terminate_child(ejabberd_sup, Ch),
+  supervisor:delete_child(ejabberd_sup, Ch),
+  ejabberd_hooks:delete(user_send_packet, Host, ?MODULE, on_user_send_packet, 100),
+  ok.
+
+on_user_send_packet(From, To, Packet) ->
+  Type = exml_query:attr(Packet, <<"type">>, <<>>),
+  Body = exml_query:path(Packet, [{element, <<"body">>}, cdata], <<>>),
+  Mod = get_callback_module(),
+  case Mod:should_make_req(Type, Body, From, To) of
```

Comment:
How about passing `Packet` instead of `Type` and `Body` to the callback function and doing whatever is need in the implementation? This would allow us to write more accurate checkers.